Common/archivereader.cpp

```diff
@@ -0,0 +1,219 @@
+//
+// Copyright (C) 2013-2018 University of Amsterdam
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+#include "archivereader.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+
+#include <sstream>
+
+#include "libzip/archive_entry.h"
+
+using namespace std;
+
+
+ArchiveReader::ArchiveReader(const string &archivePath, const string &entryPath)
+{
+	_entryPath		= entryPath;
+	_archivePath	= archivePath;
+
+	openEntry(archivePath, entryPath);
+}
+
+ArchiveReader::~ArchiveReader()
+{
+	close();
+}
+
+#ifdef _WIN32
+#define ARCHIVE_OPEN_FILENAME_FUNC archive_read_open_filename_w
+#define PATH_CONVERSION	boost::nowide::widen
+#else
+#define ARCHIVE_OPEN_FILENAME_FUNC archive_read_open_filename
+#define PATH_CONVERSION
+#endif
+
+void ArchiveReader::openEntry(const string &archivePath, const string &entryPath)
+{
+	boost::filesystem::path pathArchive = PATH_CONVERSION(archivePath);
+
+	if ((_archiveExists = boost::filesystem::exists(pathArchive)))
```

Comment:
I would happily sacrifice a line for clarity here...

Reply:
I mean, it is right above and it just becomes so ugly with those 
```
#ifdef blabla
some code
#else
almost the exact same code just begging for a typo
#endif
```

But then again, I do not see a lot of harm in macros.

Reply:
I was not talking about the MACROs, I was talking about this, `if((_archiveExists = boost::filesystem::exists(pathArchive)))`, that I have made it into two lines, 

```cpp
 _archiveExists = boost::filesystem::exists(pathArchive);

 if (_archiveExists)
```

Reply:
oow yeah, that is not a bad move indeed